lib/jobs/diag-update-firmware.js

```diff
@@ -0,0 +1,198 @@
+// Copyright 2017, Dell EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = diagUpdateFirmwareFactory;
+di.annotate(diagUpdateFirmwareFactory, new di.Provide('Job.Diag.Update.Firmware'));
+di.annotate(diagUpdateFirmwareFactory, new di.Inject(
+    'Job.Base',
+    'JobUtils.DiagTool',
+    'Logger',
+    'Util',
+    'Assert',
+    'Promise',
+    '_'
+));
+
+function diagUpdateFirmwareFactory(
+    BaseJob,
+    DiagTool,
+    Logger,
+    util,
+    assert,
+    Promise,
+    _
+) {
+    var logger = Logger.initialize(diagUpdateFirmwareFactory);
+    function DiagUpdateFirmware(options, context, taskId) {
+        DiagUpdateFirmware.super_.call(this, logger, options, context, taskId);
+
+        this.nodeId = this.context.target;
+        this.localImagePath = options.localImagePath;
+        this.imageName = options.imageName;
+        this.imageMode = options.imageMode;
+        this.firmwareName = options.firmwareName;
+        this.skipReset = options.skipReset || false;
+        this.diagImagePath = '/uploads';
```

Comment:
in previous experience,  it could be default value of task-data to avoid user to input